pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -244,31 +244,33 @@ private void serializeTree(StarTreeBuilder starTreeBuilder) throws Exception {
   private void updateTree(StarTreeIndexNode node, Map<String, HashBiMap<Object, Integer>> dictionaryMap,
```

Comment:
In the first implementation, we will first generate the existing implementation, and then convert it to the new implementation. So this part of the code will work on the existing implementation instead of the interface. Will re-visit this in the next round of changes.